src/main/java/sg/ncl/MainController.java

```diff
@@ -2417,14 +2415,82 @@ public String admin(Model model, HttpSession session) {
         model.addAttribute("userToTeamMap", userToTeamMap);
         model.addAttribute("dataList", datasetsList);
+
         return "admin2";
     }
 
+    @RequestMapping("/admin/experiments")
+    public String adminExperimentsManagement(Model model, HttpSession session) {
+
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        //------------------------------------
+        // get list of experiments
+        //------------------------------------
+        HttpEntity<String> expRequest = createHttpEntityHeaderOnly();
+        ResponseEntity expResponseEntity = restTemplate.exchange(properties.getSioExpUrl(), HttpMethod.GET, expRequest, String.class);
+
+        //------------------------------------
+        // get list of realizations
+        //------------------------------------
+        HttpEntity<String> realizationRequest = createHttpEntityHeaderOnly();
+        ResponseEntity realizationResponseEntity = restTemplate.exchange(properties.getAllRealizations(), HttpMethod.GET, realizationRequest, String.class);
+
+        JSONArray jsonExpArray = new JSONArray(expResponseEntity.getBody().toString());
+        JSONArray jsonRealizationArray = new JSONArray(realizationResponseEntity.getBody().toString());
+        Map<Experiment2, Realization> experiment2Map = new HashMap<>(); // exp id, experiment
+        Map<Long, Realization> realizationMap = new HashMap<>(); // exp id, realization
+
+        for (int k = 0; k < jsonRealizationArray.length(); k++) {
+            Realization realization = extractRealization(jsonRealizationArray.getJSONObject(k).toString());
+            if (realization.getState().equals(RealizationState.RUNNING.name())) {
+                realizationMap.put(realization.getExperimentId(), realization);
+            }
+        }
+
+        for (int i = 0; i < jsonExpArray.length(); i++) {
+            Experiment2 experiment2 = extractExperiment(jsonExpArray.getJSONObject(i).toString());
+            if (realizationMap.containsKey(experiment2.getId())) {
+                experiment2Map.put(experiment2, realizationMap.get(experiment2.getId()));
+            }
+        }
+
+        model.addAttribute("runningExpMap", experiment2Map);
+
+        return "experiment_dashboard";
+    }
+
+//    @RequestMapping(value = "/admin/{teamId}", method = RequestMethod.GET)
+//    public String admin(@PathVariable String teamId, Model model, HttpSession session) {
+//        HttpEntity<String> exprequest = createHttpEntityHeaderOnly();
+//        ResponseEntity expresponse = restTemplate.exchange(properties.getExpListByTeamId(teamId), HttpMethod.GET, exprequest, String.class);
+//        JSONArray experimentsArray = new JSONArray(expresponse.getBody().toString());
+//
+//        List<Experiment2> experimentList = new ArrayList<>();
+//        Map<Long, Realization> realizationMap = new HashMap<>();
+//
+//        for (int k = 0; k < experimentsArray.length(); k++) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)